sktime/forecasting/callbacks/callback.py

```diff
@@ -0,0 +1,38 @@
+"""Callback class for callbacks during the evaluation of a forecaster."""
+
+import pandas as pd
+
+
+class Callback:
+    """Callback class for callbacks during the evaluation of a forecaster."""
+
+    def __init__(self):
+        self._forecaster = None
+        self._scores = []
+
+    @property
+    def scores(self):
+        """Scores of the forecaster during evaluation."""
+        return self._scores
```

Comment:
I understand that your intended use case may mainly eveolve around logging evaluation metrics during tuning for different points in hyperparameter space, and in general that is certainly going to be very popular, but I am not sure if we should keep scores as part of the general schema.

At this point I do not have a alternative design suggestion, but this is just a note from my side.

Reply:
I disagree here, because precisely at the heart of "evaluating" some forecaster is using metrics and therefore what was their score. So as long as the callbacks are around evaluation i would still keep it here. I will give it some thought though. At the same time i also think that given that it is a new feature we should not over engineer it as it might backfire in the future and just provide the minimal necessary functionality.

Reply:
> precisely at the heart of "evaluating" some forecaster is using metrics and therefore what was their score

Totally agreed, but my point is callbacks does not have to be used for evaluation alone. It can be for monitoring epoch duration, it can be for monitoring exact input to the model (consider reduction forecasters or sliding windows) etc.

Reply:
I would suggest that in cases where scores are not being necessary the author of the callback implements the score setter as:

```python
@scores.setter
    def scores(self, scores):
        logger.info(f"Scores will be omitted for class self.__class__.__name__")
        self._scores =[]
```
or something similar, what do you think?